packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1976,29 +1975,7 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
         return context.realize();
     }
 
-    public setFlushMode(mode: FlushMode): void {
-        if (mode === this._flushMode) {
-            return;
-        }
-
-        this.mc.logger.sendTelemetryEvent({
-            eventName: "FlushMode Updated",
-            old: this._flushMode,
-            new: mode,
-        });
-
-        // Flush any pending batches if switching to immediate
-        if (mode === FlushMode.Immediate) {
-            this.flush();
-        }
-
-        this._flushMode = mode;
-
-        // Let the PendingStateManager know that FlushMode has been updated.
-        this.pendingStateManager.onFlushModeUpdated(mode);
-    }
-
-    public flush(): void {
+    public flush(isImmediateBatch: boolean = false): void {
```

Comment:
Is it public only because pending state manager needs to call it? Can we make it private and provide it through delegation of some sort?

Reply:
Also it would be great to have small comments on when this API is supposed to be called and what it means for this workflow.

Reply:
`flush` has been deprecated and will be removed in [AB#1076](https://dev.azure.com/fluidframework/internal/_workitems/edit/1076)